### PR TITLE
VsAppShell fixes

### PR DESCRIPTION
**Pull request by AlexanderSher**
- Added explicit `EnsureInitialize` method that is called from `RPackage`
- Added exports for `ICoreShell` and `IEditorShell`
- Removed initialization code from `VsAppShell` constructor so that it can be accessed inside constructors from any thread
- Removed `IsUITestEnvironment` that isn't used

### CODE REVIEW

File: src/Languages/Editor/Impl/Shell/IEditorShell.cs [hunk 1]
```diff
@@ -45,11 +45,6 @@ public interface IEditorShell: ICoreShell {
         bool IsUnitTestEnvironment { get; }
 
         /// <summary>
-        /// Tells if code runs in UI test environment
-        /// </summary>
-        bool IsUITestEnvironment { get; }
-
```
Comment (MikhailArkhipov) [inline src/Languages/Editor/Impl/Shell/IEditorShell.cs]: It was intended for VS-based tests originally. May come handy some day.

File: src/Package/Impl/Shell/VsAppShell.cs [hunk 2]
```diff
@@ -89,6 +99,16 @@ public sealed class VsAppShell : IApplicationShell, IMainThread, IIdleTimeServic
             }
         }
 
+        private static VsAppShell GetInstance() {
```
Comment (MikhailArkhipov) [inline src/Package/Impl/Shell/VsAppShell.cs]: So basically code can import it, but it is useless until package calls Initialize right? Maybe then at least set CompositionService and ExportProvider in GetInstance?

Comment (AlexanderSher) [inline src/Package/Impl/Shell/VsAppShell.cs]: Agreed

Comment (MikhailArkhipov) [inline src/Package/Impl/Shell/VsAppShell.cs]: Perhaps we also want assert if someone calls any method on non-initialized objects.

Comment (AlexanderSher) [inline src/Package/Impl/Shell/VsAppShell.cs]: It will fail anyway (with NRE), but I can add this verification

**Comment by MikhailArkhipov** [comment]
👍

**Comment by int19h** [comment]
:+1:

**Pull request merged**